rootio/wbuffer.go

```diff
@@ -94,20 +94,22 @@ func (w *WBuffer) WriteObjectAny(obj Object) error {
 		return w.err
 	}
 
-	//	pos := w.Pos()
-	//	w.WriteU32(0) // placeholder for bytecount.
-	//
-	//	mapsize := len(w.refs)
-	//	w.WriteClass(obj)
+	//pos := w.Pos()
+	w.WriteU32(0) // placeholder for bytecount.
 
+	mapsize := len(w.refs)
+	w.WriteClass(obj)
+	w.w.c += mapsize
 	return w.err
 }
 
 func (w *WBuffer) WriteClass(obj Object) error {
 	if w.err != nil {
 		return w.err
 	}
-	//	class := obj.Class()
+	class := obj.Class()
+	w.WriteString(class)
+	w.w.c += len(class)
```

Comment:
not sure that's really what's needed. (the failing test seems to agree with me :P)